python/runner/runner.py

```diff
@@ -177,63 +178,76 @@ async def run_instance(self, config, args):
         input_stream = Stream()
         asyncio.create_task(self.connect_input_stream(input_stream))
 
-        self.logger.info("Running instance...")
+        self.logger.info('Running instance...')
         result = self.sequence.run(context, input_stream, *args)
 
-        self.logger.info(f"Sending PANG")
+        self.logger.info(f'Sending PANG')
         monitoring = self.streams[CC.MONITORING]
-        pang_provides_data = {
+        pang_data = {
             'provides': '',
             'contentType': ''
         }
-        send_encoded_msg(monitoring, msg_codes.PANG, pang_provides_data)
+        send_encoded_msg(monitoring, msg_codes.PANG, pang_data)
+
+        try:
+            topic = self.sequence.topic()
+            pang_data = {
+                'requires': topic,
+                'contentType': 'text/plain'
+            }
+            send_encoded_msg(monitoring, msg_codes.PANG, pang_data)
+        except AttributeError:
+            self.logger.info('Sequence without topic, skipping sending PANG')
 
         if asyncio.iscoroutine(result):
             result = await result
         if result:
             await self.forward_output_stream(result)
         else:
-            self.logger.debug("Sequence returned no output.")
+            self.logger.debug('Sequence returned no output.')
 
         self.logger.info('Finished.')
 
 
     async def connect_input_stream(self, input_stream):
-        raw_headers = await self.streams[CC.IN].readuntil(b"\r\n\r\n")
-        header_list = raw_headers.decode().rstrip().split("\r\n")
-        headers = {
-            key.lower(): val for key, val in [el.split(": ") for el in header_list]
-        }
-        self.logger.info(f"Input headers: {repr(headers)}")
+        if (not hasattr(self.sequence, 'topic')):
```

Comment:
I understand that when input comes from a topic it doesn't have headers, but is that expected or is it a bug? Does it always behave in this way? I think we need a confirmation from Ala or someone else more familiar with how sth works.

Reply:
@alicja-gruzdz what do you think?

Reply:
The topics currently only support `application/x-ndjson`. However, in the later stages we should support other content types.
I checked by running the Sequences that require and provide the same topic:
https://github.com/scramjetorg/transform-hub/tree/release/0.16/packages/reference-apps/delay-out-in-a
https://github.com/scramjetorg/transform-hub/tree/release/0.16/packages/reference-apps/delay-out-in-b
and the information about content type came in the headers to `delay-out-in-b`, which requires the topic:
![Zrzut ekranu z 2022-02-21 10-06-59](https://user-images.githubusercontent.com/8177865/154923540-36738d73-7e0b-4887-85a2-9a2eae78104c.png)

Reply:
I talked with @pietrzakacper and he confirmed that there is no way to get headers in this case, in tsrunner headers are retrieved from the sequence

Reply:
Sorry, you're right that in the case of `requires` it sets https://github.com/scramjetorg/transform-hub/blob/release/0.17/packages/runner/src/runner.ts#L336

Reply:
Yes, we have two sepeare use cases for determining content type of the input stream:
1. User writes to sequence input through REST API (or CLI on top of it). In this case the[ input stream will receive  headers](https://github.com/scramjetorg/transform-hub/blob/release/0.17/packages/runner/src/runner.ts#L351) containing `Content-Type`.
2. Sequence requires a topic as an input stream. In this case [content-type is specified by the Sequence](https://github.com/scramjetorg/transform-hub/blob/release/0.17/packages/runner/src/runner.ts#L336) (`contentType` field next to `requires`). No headers will be sent on the input stream, just the data that sits on the topic.